src/Symfony/Component/Mailer/Event/MessageEvent.php

```diff
@@ -64,4 +68,17 @@ public function isQueued(): bool
     {
         return $this->queued;
     }
+
+    public function addStamp(StampInterface $stamp): void
+    {
```

Comment:
We should probably throw an exception when `queued` is `false` as these stamps won't be used at all in that case.

Reply:
done